lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -236,6 +266,27 @@ struct ConversionOpConversion : public OpConversionPattern<ConversionOp> {
 // Statement Conversion
 //===----------------------------------------------------------------------===//
 
+struct AssignOpConversion : public OpConversionPattern<BlockingAssignOp> {
```

Comment:
I think `moore.blocking_assgin` and `moore.assign` both use this conversion pattern. I just want to know whether this conversion is reasonable. And `moore.assign` has a parent--`moore.module`, so I use `moore.blocking_assign` to demonstrate. For `moore.nonblocking_assign`, I think we can first create a `hw.wire`, then when we handle `moore.nonblocking_assign`, then we can also replace the `hw.wire` have been created using `seq.firreg`.  Like this code fragment:   // It's similar to the following context handling assignments.
```
auto name =
              dyn_cast<VariableOp>(op.getDst().getDefiningOp()).getNameAttr();
          auto regOp = rewriter.create<seq::FirRegOp>(procedureOp->getLoc(),
                                                      input, clk, name);

          // Update the variables.
          auto from = rewriter.getRemappedValue(op.getDst());
          rewriter.replaceOp(from.getDefiningOp(), regOp);
```